### PR TITLE
SNOW-743920: Add test fetching many chunks

### PR DESCRIPTION
**Pull request by sfc-gh-dprzybysz**
### Description
Add test for specific chunk download performance problem

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: test/integration/testLargeResultSet.js [hunk 1]
```diff
@@ -106,3 +107,43 @@ describe('Large result Set Tests', function ()
     });
   });
 });
+
+describe('SNOW-743920: Large result set with ~35 chunks', function () {
+  let connection;
+  const tableName = 'test_table';
+  const sourceRowCount = 251002;
```
Comment (sfc-gh-pfus) [inline test/integration/testLargeResultSet.js:114]: Why such random number?

File: test/integration/testLargeResultSet.js [hunk 2]
```diff
@@ -106,3 +107,43 @@ describe('Large result Set Tests', function ()
...
+  const createTable = `create or replace table ${tableName} (data string)`;
+  const populateData = `insert into ${tableName} select randstr(20, random()) from table (generator(rowcount =>${sourceRowCount}))`;
+  const selectAllFromOrders = `select * from ${tableName}`;
```
Comment (sfc-gh-pfus) [inline test/integration/testLargeResultSet.js:117]: What do you mean by `orders`?

File: test/integration/testLargeResultSet.js [hunk 3]
```diff
@@ -106,3 +107,43 @@ describe('Large result Set Tests', function ()
...
+
+  before(async () => {
+    connection = testUtil.createConnection();
+    await testUtil.connectAsync(connection);
+    await testUtil.executeCmdAsync(connection, 'alter session set ROWS_PER_RESULTSET = 1000000');
```
Comment (sfc-gh-pfus) [inline test/integration/testLargeResultSet.js:122]: This result set is larger than the number of source rows, is it correct?

File: test/integration/testLargeResultSet.js [hunk 4]
```diff
@@ -106,3 +107,43 @@ describe('Large result Set Tests', function ()
...
+    await testUtil.executeCmdAsync(connection, 'alter session set USE_CACHED_RESULT = false;');
+    await testUtil.executeCmdAsync(connection, createTable);
+    await testUtil.executeCmdAsync(connection, populateData);
+    configureLogger('TRACE');
+  });
+
+  after(async () => {
+    configureLogger('ERROR');
```
Comment (sfc-gh-pfus) [inline test/integration/testLargeResultSet.js:130]: Does it affect other tests?

File: test/integration/testLargeResultSet.js [hunk 5]
```diff
@@ -106,3 +107,43 @@ describe('Large result Set Tests', function ()
...
+    await testUtil.dropTablesIgnoringErrorsAsync(connection, [tableName]);
```
Comment (sfc-gh-pfus) [inline test/integration/testLargeResultSet.js:131]: What if this line ends with error? Will the connection be closed?

File: test/integration/testLargeResultSet.js (hunk 1 above)
Comment (sfc-gh-dprzybysz) [reply test/integration/testLargeResultSet.js:258]: It's connected with tested issue

File: test/integration/testLargeResultSet.js (hunk 2 above)
Comment (sfc-gh-dprzybysz) [reply test/integration/testLargeResultSet.js]: it's already renamed to just `selectData`

File: test/integration/testLargeResultSet.js (hunk 3 above)
Comment (sfc-gh-dprzybysz) [reply test/integration/testLargeResultSet.js:266]: it's also connected with reported issue - but it has to be higher than number of downloaded rows

File: test/integration/testLargeResultSet.js (hunk 4 above)
Comment (sfc-gh-dprzybysz) [reply test/integration/testLargeResultSet.js:273]: ERROR is sane logging level in tests we should keep - eventually this logger will be removed from this test when analysis will be finished

File: test/integration/testLargeResultSet.js (hunk 5 above)
Comment (sfc-gh-dprzybysz) [reply test/integration/testLargeResultSet.js:274]: it shouldn't end with error as it ignores errors :)

**Review by sfc-gh-pfus — APPROVED** [review]

File: lib/connection/result/result.js [hunk 6]
```diff
@@ -9,8 +9,8 @@ var Chunk = require('./chunk');
 var ResultStream = require('./result_stream');
 var ChunkCache = require('./chunk_cache');
 var Column = require('./column');
-var Parameters = require('../../parameters');
```
Comment (sfc-gh-igarish) [inline lib/connection/result/result.js:12]: is it unused? why we removed it?

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

File: lib/connection/result/result.js (hunk 6 above)
Comment (sfc-gh-dprzybysz) [reply lib/connection/result/result.js:12]: yes, it's unused

**Pull request merged**